packages/cw3/src/helpers.rs

```diff
@@ -0,0 +1,80 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{to_binary, Api, CanonicalAddr, CosmosMsg, HumanAddr, StdResult, WasmMsg};
+
+use crate::msg::{Cw3HandleMsg, Vote};
+use cw0::Expiration;
+
+/// Cw3Contract is a wrapper around HumanAddr that provides a lot of helpers
+/// for working with this.
+///
+/// If you wish to persist this, convert to Cw3CanonicalContract via .canonical()
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Cw3Contract(pub HumanAddr);
+
+impl Cw3Contract {
+    pub fn addr(&self) -> HumanAddr {
+        self.0.clone()
+    }
+
+    /// Convert this address to a form fit for storage
+    pub fn canonical<A: Api>(&self, api: &A) -> StdResult<Cw3CanonicalContract> {
+        let canon = api.canonical_address(&self.0)?;
+        Ok(Cw3CanonicalContract(canon))
+    }
+
+    pub fn encode_msg(&self, msg: Cw3HandleMsg) -> StdResult<CosmosMsg> {
```

Comment:
- This does not support custom message types
- If we always return a `WasmMsg`, why not return `StdResult<WasmMsg>`. This would remove all the issues with a generic result type

Reply:
I will leave a TODO here to make it more generic and fix it when using it.
That will only happen with an on-chain cw3 consumer.